crates/common_utils/src/id_type/tenant.rs

```diff
@@ -0,0 +1,30 @@
+use super::{AlphaNumericId, LengthId};
+use crate::errors::{CustomResult, ValidationError};
+
+crate::id_type!(
+    TenantId,
+    "A type for tenant_id that can be used for unique identifier for a tenant"
+);
+crate::impl_id_type_methods!(TenantId, "tenant_id");
+
+// This is to display the `TenantId` as TenantId(abcd)
+crate::impl_debug_id_type!(TenantId);
+crate::impl_try_from_cow_str_id_type!(TenantId, "tenant_id");
+
+crate::impl_serializable_secret_id_type!(TenantId);
+crate::impl_queryable_id_type!(TenantId);
+crate::impl_to_sql_from_sql_id_type!(TenantId);
+
+impl TenantId {
+    /// Get tenant id from String
+    pub fn wrap(tenant_id: String) -> CustomResult<Self, ValidationError> {
```

Comment:
Nit: Can we use a better name which indicates what it's being constructed from, say `try_from_string()` maybe?

Reply:
Yes we can, for org id we follow the `wrap()`, i just thought to keep things consistent. Shall i change it there as well?

Reply:
Let it remain for now, don't take that change in this PR.

Also looping in @sai-harsha-vardhan and @Narayanbhat166 about this.